tsl/src/nodes/vector_agg/plan.c

```diff
@@ -215,13 +215,13 @@ is_vector_var(const VectorQualInfo *vqinfo, Expr *expr)
 
 	Var *var = castNode(Var, expr);
 
-	if (var->varattno <= 0)
+	if (var->varattno <= 0 || var->varattno > vqinfo->num_attrs)
```

Comment:
Now it looks like `num_attrs` can be meaningfully different, I started wondering when that happens and then saw that this is for assertions only. Maybe makes sense to make it an assertion instead?

Reply:
Yes, I can make it an assertion.